subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -0,0 +1,630 @@
+	scan->per_scan.node_rx_estab = node_rx;
+	sync->node_rx_lost.link = link_sync_lost;
+
+	/* Initialise ULL and LLL headers */
+	ull_hdr_init(&sync->ull);
+	lll_hdr_init(lll_sync, sync);
+
+	/* Enable scanner to create sync */
+	scan->per_scan.sync = sync;
+	if (IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED)) {
+		scan_coded->per_scan.sync = sync;
+	}
+
+	return 0;
+}
+
+uint8_t ll_sync_create_cancel(void **rx)
+{
+	struct ll_scan_set *scan_coded;
+	memq_link_t *link_sync_estab;
+	memq_link_t *link_sync_lost;
+	struct node_rx_pdu *node_rx;
+	struct ll_scan_set *scan;
+	struct ll_sync_set *sync;
+	struct node_rx_sync *se;
+
+	scan = ull_scan_set_get(SCAN_HANDLE_1M);
+	if (!scan || !scan->per_scan.sync) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED)) {
+		scan_coded = ull_scan_set_get(SCAN_HANDLE_PHY_CODED);
+		if (!scan_coded || !scan_coded->per_scan.sync) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+	}
+
+	sync = scan->per_scan.sync;
+	scan->per_scan.sync = NULL;
+	if (IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED)) {
+		scan_coded->per_scan.sync = NULL;
+	}
+
+	/* Check for race condition where in sync is established when sync
+	 * context was set to NULL.
+	 */
+	if (sync->timeout_reload) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	node_rx = (void *)scan->per_scan.node_rx_estab;
+	link_sync_estab = node_rx->hdr.link;
+	link_sync_lost = sync->node_rx_lost.link;
+
+	ll_rx_link_release(link_sync_lost);
+	ll_rx_link_release(link_sync_estab);
+	ll_rx_release(node_rx);
+
+	node_rx = (void *)&sync->node_rx_lost;
+	node_rx->hdr.type = NODE_RX_TYPE_SYNC;
+	node_rx->hdr.handle = 0xffff;
+	node_rx->hdr.rx_ftr.param = sync;
+	se = (void *)node_rx->pdu;
+	se->status = BT_HCI_ERR_OP_CANCELLED_BY_HOST;
+
+	*rx = node_rx;
+
+	return 0;
+}
+
+uint8_t ll_sync_terminate(uint16_t handle)
+{
+	memq_link_t *link_sync_lost;
+	struct ll_sync_set *sync;
+	uint32_t volatile ret_cb;
+	uint32_t ret;
+	void *mark;
+
+	sync = is_enabled_get(handle);
+	if (!sync) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	mark = ull_disable_mark(sync);
+	LL_ASSERT(mark == sync);
+
+	ret_cb = TICKER_STATUS_BUSY;
+	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_THREAD,
+			  TICKER_ID_SCAN_SYNC_BASE + handle,
+			  ull_ticker_status_give, (void *)&ret_cb);
+	ret = ull_ticker_status_take(ret, &ret_cb);
+	if (ret) {
+		mark = ull_disable_mark(sync);
+		LL_ASSERT(mark == sync);
+
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	mark = ull_disable_unmark(sync);
+	LL_ASSERT(mark == sync);
+
+	link_sync_lost = sync->node_rx_lost.link;
+	ll_rx_link_release(link_sync_lost);
+
+	ull_sync_release(sync);
+
+	return 0;
+}
+
+uint8_t ll_sync_recv_enable(uint16_t handle, uint8_t enable)
+{
+	/* TODO: */
+	return BT_HCI_ERR_CMD_DISALLOWED;
+}
+
+int ull_sync_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int ull_sync_reset(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
```

Comment:
Can we replace with 'return err;'?

Reply:
as previously explained, this is style considering future additions of code that may follow after init_reset. No action to take.